src/map/lapack2flash/FLASH_trtri.c

```diff
@@ -0,0 +1,174 @@
+/*
+
+    Copyright (C) 2014, The University of Texas at Austin
+
+    This file is part of libflame and is available under the 3-Clause
+    BSD license, which can be found in the LICENSE file at the top-level
+    directory, or at http://opensource.org/licenses/BSD-3-Clause
+
+*/
+
+#include "FLAME.h"
+
+#ifdef FLA_ENABLE_LAPACK2FLASH
+
+
+#include "FLASH_lapack2flash_util_defs.h"
+#include "FLA_lapack2flame_return_defs.h"
+#include "FLA_lapack2flame_prototypes.h"
+
+/*
+   TRTRI computes the inverse of a upper or lower triangular
+   matrix A.
+
+   This is the Level 3 BLAS version of the algorithm.
+
+   INFO
+   = 0: successful exit
+   < 0: if INFO = -i, the i-th argument had an illegal value -> LAPACK_trtri_op_check
+   > 0: if INFO = i, A(i,i) is exactly zero. The triangular -> LAPACK_check_diag_zero
+   matrix is singular and its inverse can not be computed.
+
+   Hence, if the routine passes above error checking, FLA_Trinv should not produce any error.
+*/
+
+#define LAPACK_trtri(prefix)                                            \
+  int F77_ ## prefix ## trtri( char* uplo,                              \
+                               char* diag,                              \
+                               int* n,                                  \
+                               PREFIX2LAPACK_TYPEDEF(prefix)* buff_A,   \
+                               int* ldim_A,                             \
+                               int* info )
+
+#define LAPACK_trtri_body(prefix)                               \
+  FLA_Datatype datatype = PREFIX2FLAME_DATATYPE(prefix);        \
+  FLA_Uplo     uplo_fla;                                        \
+  FLA_Diag     diag_fla;                                        \
+  FLA_Obj      A;                                               \
+  FLA_Error    init_result;                                     \
+  dim_t        blocksize = FLASH_Queue_get_hip_blocksize();     \
+                                                                \
+  FLA_Init_safe( &init_result );                                \
+                                                                \
+  FLA_Param_map_netlib_to_flame_uplo( uplo, &uplo_fla );        \
+  FLA_Param_map_netlib_to_flame_diag( diag, &diag_fla );        \
+                                                                \
+  FLASH_Obj_create_without_buffer( datatype,                    \
+                                   *n,                          \
+                                   *n,                          \
+                                   FLASH_Queue_get_hip_depth(), \
+                                   &blocksize,                  \
```

Comment:
yeah - that's the awkward part with "HIP" :-D

Reply:
Updated the names! https://github.com/iotamudelta/libflame/commit/926e05ee52bc9f34e87d35af34eb71042ab6b6ed

Reply:
like it!